src/Components/Components/src/Routing/RoutingStateProvider.cs

```diff
@@ -0,0 +1,15 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace Microsoft.AspNetCore.Components.Routing;
+
+/// <summary>
+/// Provides RouteData
+/// </summary>
+public abstract class RoutingStateProvider
+{
+    /// <summary>
+    /// Gets or sets RouteData
+    /// </summary>
+    public RouteData? RouteData { get; set; }
```

Comment:
Is there a scenario why someone else (other than the framework) would need to set this value?

Reply:
Most likely no

Reply:
If that's the case, should this be settable from outside then?

Reply:
Yes, because I wouldn't be able to set `RouteData` in `EndpointHtmlRenderer.InitializeStandardComponentServicesAsync`.

Reply:
I don't think there are alternatives. `RoutingStateProvider.RouteData` is set in `Components.Endpoints.EndpointHtmlRenderer` and then I need to get the value in `Components.Router` component.

Reply:
> I don't think there are alternatives. RoutingStateProvider.RouteData is set in Components.Endpoints.EndpointHtmlRenderer and then I need to get the value in Components.Router component.

I think there are, for example, how does this work in `NavigationManager`?